test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp

```diff
@@ -0,0 +1,137 @@
+/* Copyright 2021 Andrea Bocci
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// make sure the CPU_B_SEQ_T_SEQ backend is always available
+#ifndef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#    define ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#endif // ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#include <alpaka/alpaka.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+#include <iostream>
+
+constexpr int memset_value(int c)
+{
+    int v = c & 0xFF;
+    return (v | v << 8 | v << 16 | v << 24);
+}
+
+// 0- and 1- dimensional space
+using Idx = std::size_t;
+using Dim0D = alpaka::DimInt<0u>;
+using Dim1D = alpaka::DimInt<1u>;
+using Scalar = alpaka::Vec<Dim0D, Idx>;
+
+// enabled accelerators with 1-dimensional kernel space
+using TestAccs = alpaka::test::EnabledAccs<Dim1D, Idx>;
+
+TEMPLATE_LIST_TEST_CASE("zeroDimBuffer", "[zeroDimBuffer]", TestAccs)
+{
+    using DeviceAcc = TestType;
+    using Device = alpaka::Dev<DeviceAcc>;
+    using DeviceQueue = alpaka::Queue<DeviceAcc, alpaka::NonBlocking>;
+
+    using HostAcc = alpaka::AccCpuSerial<Dim1D, Idx>;
+    using Host = alpaka::DevCpu;
+    using HostQueue = alpaka::Queue<HostAcc, alpaka::Blocking>;
+
+    // check that a Scalar extent has exactly 1 element
+    Scalar scalar;
+    INFO("Scalar size: " << scalar.prod())
+
+    // CPU host
+    auto const host = alpaka::getDevByIdx<Host>(0u);
+    INFO("Using alpaka accelerator: " << alpaka::getAccName<HostAcc>())
+    HostQueue hostQueue(host);
+
+    // host buffer
+    auto h_buffer = alpaka::allocBuf<int, Idx>(host, Scalar{});
+    INFO(
+        "host buffer allocated at " << alpaka::getPtrNative(h_buffer) << " with "
+                                    << alpaka::extent::getExtentProduct(h_buffer) << " elements")
+
+    // async host buffer
+    auto h_async = alpaka::allocAsyncBuf<int, Idx>(hostQueue, Scalar{});
+    INFO(
+        "async host buffer allocated at " << alpaka::getPtrNative(h_async) << " with "
+                                          << alpaka::extent::getExtentProduct(h_async) << " elements")
+
+    // host-side buffer memset
+    const int value1 = 42;
+    INFO("host-side buffer memset")
+    alpaka::memset(hostQueue, h_buffer, value1, Scalar{});
+    alpaka::wait(hostQueue);
+    INFO("check: " << *alpaka::getPtrNative(h_buffer) << " == " << memset_value(value1))
+    REQUIRE(memset_value(value1) == *alpaka::getPtrNative(h_buffer));
```

Comment:
```suggestion
    CHECK(memset_value(value1) == *alpaka::getPtrNative(h_buffer));
```
Unless you really need the test to abort when this is wrong, use `CHECK` please.

Reply:
Personally, I don't need the test at all. I have been asked to write one, without any indications or instructions as to how to do it, and this is what I could came up with.

I **strongly** suggest you start working on some documentation on how to add tests and run the existing ones, if you are interested in have any contributions to Alpaka.

Reply:
Anyway, sure, I can replace everywhere
```c++
    INFO("check: " << *alpaka::getPtrNative(h_buffer) << " == " << memset_value(value1))
    REQUIRE(memset_value(value1) == *alpaka::getPtrNative(h_buffer));
```
with
```c++
    CHECK(memset_value(value1) == *alpaka::getPtrNative(h_buffer));
```